opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/SimpleQueryString.java

```diff
@@ -0,0 +1,123 @@
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> fuzzyPrefixLength =
+      (b, v) -> b.fuzzyPrefixLength(Integer.parseInt(v.stringValue()));
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> fuzzyTranspositions =
+      (b, v) -> b.fuzzyTranspositions(Boolean.parseBoolean(v.stringValue()));
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> lenient =
+      (b, v) -> b.lenient(Boolean.parseBoolean(v.stringValue()));
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> minimumShouldMatch =
+      (b, v) -> b.minimumShouldMatch(v.stringValue());
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> quoteFieldSuffix =
+      (b, v) -> b.quoteFieldSuffix(v.stringValue());
+  private final BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder> boost =
+      (b, v) -> b.boost(Float.parseFloat(v.stringValue()));
+
+  ImmutableMap<Object, Object> argAction = ImmutableMap.builder()
+      .put("analyze_wildcard", analyzeWildcard)
+      .put("analyzer", analyzer)
+      .put("auto_generate_synonyms_phrase_query", autoGenerateSynonymsPhraseQuery)
+      .put("flags", flags)
+      .put("fuzzy_max_expansions", fuzzyMaxExpansions)
+      .put("fuzzy_prefix_length", fuzzyPrefixLength)
+      .put("fuzzy_transpositions", fuzzyTranspositions)
+      .put("lenient", lenient)
+      .put("default_operator", defaultOperator)
+      .put("minimum_should_match", minimumShouldMatch)
+      .put("quote_field_suffix", quoteFieldSuffix)
+      .put("boost", boost)
+      .build();
+
+  @Override
+  public QueryBuilder build(FunctionExpression func) {
+    Iterator<Expression> iterator = func.getArguments().iterator();
+    NamedArgumentExpression fields = (NamedArgumentExpression) iterator.next();
+    NamedArgumentExpression query = (NamedArgumentExpression) iterator.next();
+    SimpleQueryStringBuilder queryBuilder = QueryBuilders.simpleQueryStringQuery(
+        query.getValue().valueOf(null).stringValue());
+    queryBuilder.fields(parseFields(fields.getValue().valueOf(null).stringValue()));
+    while (iterator.hasNext()) {
+      NamedArgumentExpression arg = (NamedArgumentExpression) iterator.next();
+      if (!argAction.containsKey(arg.getArgName())) {
+        throw new SemanticCheckException(String
+            .format("Parameter %s is invalid for simple_query_string function.", arg.getArgName()));
+      }
+      ((BiFunction<SimpleQueryStringBuilder, ExprValue, SimpleQueryStringBuilder>) argAction
+          .get(arg.getArgName()))
+          .apply(queryBuilder, arg.getValue().valueOf(null));
+    }
+    return queryBuilder;
+  }
+
+  private Map<String, Float> parseFields(String fields) {
+    try {
+      // TODO support elements wrapped by single quotes
+      var arr = new JSONArray(fields);
```

Comment:
Teh JSON parser doesn't work for 2 reasons: 
1) it doesn't support single quotes
2) it doesn't parse the carot (^) character.  
We should consider abandoning this direction and using the lex parser to read the array.